presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -149,15 +149,15 @@ public PlanNode visitTableFinish(TableFinishNode node, RewriteContext<Context> c
             // TODO: begin these operations in pre-execution step, not here
             if (target instanceof TableWriterNode.CreateName) {
                 TableWriterNode.CreateName create = (TableWriterNode.CreateName) target;
-                return new TableWriterNode.CreateHandle(metadata.beginCreateTable(session, create.getCatalog(), create.getTableMetadata(), create.getLayout()));
+                return new TableWriterNode.CreateHandle(metadata.beginCreateTable(session, create.getCatalog(), create.getTableMetadata(), create.getLayout()), create.getTableMetadata().getTable());
             }
             if (target instanceof TableWriterNode.InsertReference) {
                 TableWriterNode.InsertReference insert = (TableWriterNode.InsertReference) target;
-                return new TableWriterNode.InsertHandle(metadata.beginInsert(session, insert.getHandle()));
+                return new TableWriterNode.InsertHandle(metadata.beginInsert(session, insert.getHandle()), metadata.getTableMetadata(session, insert.getHandle()).getTable());
```

Comment:
This is a bit of a hack. We shouldn't have to store this information in the handle. It should be possible to get it by asking the metadata. However, the way BeginTableWrite works is already somewhat hacky, so it's probably ok to leave as is for now. Just add a TODO so we know why it's being done this way.